### PR TITLE
[opt](bloomfilter index) optimize memory usage for bloom filter index writer

### PR DESCRIPTION
**Pull request by airborne12**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #xxx

Problem Summary:

Optimize memory usage when adding string values for bloom filter index.
Using uint64 hash value instead of string values itself, it is expected to save a lot of memory for especially long text

### Release note

None

### Check List (For Author)

- Test 
    - [ ] Regression test
    - [x] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [x] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by hello-stephen** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by airborne12** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/bloom_filter.h [hunk 1]
```diff
@@ -167,6 +167,16 @@ class BloomFilter {
         return hash_code;
     }
 
+    static Result<uint64_t> hash(const char* buf, uint32_t size, HashStrategyPB strategy) {
+        if (strategy == HASH_MURMUR3_X64_64) {
+            uint64_t hash_code;
+            murmur_hash3_x64_64(buf, size, DEFAULT_SEED, &hash_code);
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/bloom_filter.h:173]: use _hash_func

File: be/src/olap/rowset/segment_v2/bloom_filter.h [hunk 2]
```diff
@@ -167,6 +167,16 @@ class BloomFilter {
         return hash_code;
     }
 
+    static Result<uint64_t> hash(const char* buf, uint32_t size, HashStrategyPB strategy) {
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/bloom_filter.h:170]: can you use the function: hash(const char* buf, uint32_t size)?

**Comment by airborne12** [comment]
run buildall

**Review by qidaye — DISMISSED** [review]
LGTM

**Comment by airborne12** [comment]
run buildall

**Review by zzzxl1993 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]

**Pull request merged**